### PR TITLE
fix: Add async and default error handler

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR adds an async handler wrapper that catches errors in async handlers and calls the next function so the errors can be  handled by the express error handlers. A simple error handler is added too, replicating the error handler in the decentraland server code.

### CODE REVIEW

File: src/common/ErrorHandling.spec.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import { asyncHandler } from './asyncHandler'
+
+const errorMessage = 'anErrorMessage'
+const errorStatusCode = 400
+const errorData = { id: 'anId' }
+
+const handlerWithHttpError = () => {
+  throw new HTTPError(errorMessage, errorData, errorStatusCode)
+}
+const handlerWithCommonError = () => {
```
Comment (fzavalia) [inline src/common/ErrorHandling.spec.ts:15]: Are both handlerWithHttpError and handlerWithCommonError supposed to throw an HTTPError?

Comment (LautaroPetaccio) [reply src/common/ErrorHandling.spec.ts:15]: They're not! I'll fix those issues

Comment (LautaroPetaccio) [reply src/common/ErrorHandling.spec.ts:15]: Done, fixed.

File: src/common/ErrorHandling.spec.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+          ok: false,
+        })
+      })
+  })
+})
+
+describe('when handling sync http errors', () => {
+  it('should respond with the a 500 an the error message', () => {
+    return server
+      .get(buildURL('/syncError'))
```
Comment (fzavalia) [inline src/common/ErrorHandling.spec.ts:82]: This endpoint does not throw an http error, just a common error (testing the same as the next test)

File: src/common/ExpressApp.ts [hunk 3]
```diff
@@ -118,7 +118,7 @@ export class ExpressApp {
     return this
   }
 
-  use(...handers: express.RequestHandler[]) {
+  use(...handers: express.RequestHandler[] | express.ErrorRequestHandler[]) {
```
Comment (fzavalia) [inline src/common/ExpressApp.ts:121]: typo in handers -> handlers

**Review by fzavalia — APPROVED** [review]
LGTM!

File: src/common/ErrorHandling.spec.ts (hunk 2 above)
Comment (LautaroPetaccio) [reply src/common/ErrorHandling.spec.ts]: Changed to syncHttpError!

File: src/common/ExpressApp.ts (hunk 3 above)
Comment (LautaroPetaccio) [reply src/common/ExpressApp.ts:121]: Fixed it!

**Pull request merged**